### ISSUE TITLE
Terraform 0.12 shows warnings only when there are changes

### ISSUE THREAD

**Issue by cthulhu** [hashicorp/terraform#22541]
### Terraform Version

```
terraform version
Terraform v0.12.6
```

### Terraform Configuration Files

main.tf
```hcl
provider "null" {

}
resource "null_resource" "res" {
}
```
main.tfvars
```hcl
test_var="12345"
```
### Run
```bash
terraform init && terraform plan -var-file=main.tfvars
```

### Expected Behavior

Warnings should be shown even if there are no changes.

### Actual Behavior

Warnings are shown only when there are changes.

### Steps to Reproduce

1. `terraform init`
2. `terraform plan -var-file=main.tfvars`

Changes and warnings are shown

1. `terraform apply -var-file=main.tfvars -auto-approve`
2. `terraform plan -var-file=main.tfvars`

No changes, no warnings

### Additional Context

After migration to terraform 0.12 we saw stricter behavior and warnings we would like to address, but we see them only when there are changes. So we have to create a change artificially in order to fix those recommendations. 

Example of the warning:

```
Warning: Value for undeclared variable

  on main.tfvars line 1:
   1: test_var="12345"

The root module does not declare a variable named "test_var". To use this
value, add a "variable" block to the configuration.

Using a variables file to set an undeclared variable is deprecated and will
become an error in a future release. If you wish to provide certain "global"
settings to all configurations in your organization, use TF_VAR_...
environment variables to set these instead.
```

**Comment by wazcov** [comment]
+1

**Comment by nyurik** [comment]
This is a serious bug, because some providers, e.g. Google, rely on this functionality - the [3.x upgrade guide](https://www.terraform.io/docs/providers/google/guides/version_3_upgrade.html#provider-version-configuration):

```
Before upgrading to version 3.0.0, it is recommended to upgrade to the most recent 2.X series release of the provider, make the changes noted in this guide, and ensure that your environment successfully runs terraform plan without unexpected changes or deprecation notices.
```

Now I have to keep faking some changes just to show the deprecation warnings in `terraform plan`

**Comment by nyurik** [comment]
I hacked a fix #24095 for this bug, hopefully it would solve the problem for everyone.
